### PR TITLE
RD-5834-define-credentials-in-azure-plugin-yaml

### PR DESCRIPTION
**Pull request by Nelynehemia**

### CODE REVIEW

File: cloudify_azure/utils.py [hunk 1]
```diff
@@ -440,7 +440,9 @@ def handle_properties_keys(data):
     return cleanup_empty_params(data)
 
 
-def get_client_config(properties):
+def get_client_config(properties, other_properties=None):
+    other_properties = other_properties or {}
+    properties.update(other_properties)
```
Comment (EarthmanT) [inline cloudify_azure/utils.py:445]: I prefer that we will override other properties with properties. Order of importance is:
1. Client Config from Node Props
2. Client Config from Plugin Props
That means that we first get plugin props client config and then if there is a client config from node props, we override the first with that.

```suggestion
    other_properties.update(properties)
```

**Review by EarthmanT — CHANGES_REQUESTED** [review]

File: cloudify_azure/workflows/tests/test_discover.py [hunk 2]
```diff
@@ -171,6 +171,7 @@ def test_get_resources(self, cluster_list, *_, **__):
         resource_2 = MagicMock(id='foo', location='region2')
         resource_2.as_dict.return_value = {'foo': 'bar'}
         cluster_list.return_value = [resource_1, resource_2]
+        print
```
Comment (ahmadiesa-abu) [inline cloudify_azure/workflows/tests/test_discover.py:174]: let's remove this print from here

File: cloudify_azure/utils.py [hunk 3]
```diff
@@ -441,9 +441,23 @@ def handle_properties_keys(data):
...
 
+    final_client_config = {}
+    if plugin_props:
+        for k, v in plugin_props.items():
+            if isinstance(v, dict):
+                final_client_config[k] = v.get('value')
+            else:
+                final_client_config[k] = v
+        client_config.update(final_client_config)
+        azure_config.update(final_client_config)
```
Comment (ahmadiesa-abu) [inline cloudify_azure/utils.py:459]: here we don't want to override the node level properties from the plugin properties level , it should be the other way around [ we can override the plugin properties from the node level ]

**Comment by ahmadiesa-abu** [comment]
and one more thing , we need to bump the version after fixing all the tests [ and change log ]

File: cloudify_azure/utils.py [hunk 4]
```diff
@@ -457,8 +469,12 @@ def get_client_config(properties):
     if len(azure_config_keys) > len(client_config_keys):
         ctx.logger.warn("azure_config is deprecated please use client_config, "
                         "in later version it will be removed")
-        return azure_config
-    return client_config
+
+        final_client_config.update(azure_config)
```
Comment (EarthmanT) [inline cloudify_azure/utils.py:473]: We should double check that we are not overwriting values with empty ones. For example, let's say we have this:

```
azure_config = {
  'key': None,
  'key2': 'foo'
}

client_config = {
  'key': 'foobar',
}

```

Then `{'key': 'foobar'}` will be overwritten by `{'key': None}`. This is not what we want.

**Pull request merged**